daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/RunnerFactory.scala

```diff
@@ -91,75 +98,127 @@ object Runner {
  * test can be run in parallel.
  *
  * Note however, that each thread will get its own copy of the DFDLTestSuite
+ *
+ * source is either a scala.xml.Elem, generally used for unit testing or a
+ * String. If the string starts with a / is it treated is if it were resoucrce
+ * on the classpath. Otherwise it is treated as if it were a URI.
  */
-class Runner private (elem: scala.xml.Elem, dir: String, file: String,
+class Runner private (
+  source: Either[scala.xml.Elem, String],
   validateTDMLFile: Boolean = true,
   validateDFDLSchemas: Boolean = true,
   compileAllTopLevel: Boolean = false,
   defaultRoundTripDefault: RoundTrip = Runner.defaultRoundTripDefaultDefault,
   defaultValidationDefault: String = Runner.defaultValidationDefaultDefault,
   defaultImplementationsDefault: Seq[String] = Runner.defaultImplementationsDefaultDefault) {
 
-  /*
-   * these constructors are for use by Java programs
+
+  /**
+   * Create a runner for the path. This constructor requires this path be on
+   * the classpath. A forward slash is prepended to ensure this path aways
+   * works as a classpath resource.
    */
-  def this(dir: String, file:String) =
-    this(null, dir, file)
+  def this(path: String) =
+    this(
+      if (path.startsWith("/")) Right(path)
+      else Right("/" + path)
```

Comment:
Hmmm. Not sure I would add tests showing this auto pre-slashifies the path. If this isn't covered I think I'd rather change this to assert that the path must startwith "/" and fail otherwise. 

OTOH. I guess others may have TDML files that depend on this behavior. So I guess a test would be good. You can just modify some other test this way, and have that test also exercise this little feature.

Reply:
Yeah, It's very possible other DFDL projects might hit this. It wouldn't be too hard to add some tests to make sure all these constructors are covered and work correctly. Will add.